helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/ConstraintBasedAlgorithm.java

```diff
@@ -170,10 +176,8 @@ private double getAssignmentNormalizedScore(AssignableNode node, AssignableRepli
             // Note that to ensure the algorithm is deterministic with the same inputs, do not use
             // Random functions here. Use hashcode based on the cluster topology information to get
             // a controlled randomized order is good enough.
-            Long replicaHash1 = (long) Objects
-                .hash(replica1.toString(), clusterModel.getAssignableNodes().keySet());
-            Long replicaHash2 = (long) Objects
-                .hash(replica2.toString(), clusterModel.getAssignableNodes().keySet());
+            Integer replicaHash1 = replicaHashCodeMap.get(replica1.toString());
```

Comment:
nit: primitive int

Reply:
I like the compareTo method with Integer.

Reply:
Integer class will possibly throw NPE and that's generally not a good idea using it .

Reply:
The fact is that I need to use the map for avoiding duplicate calculating. If I use map, I have to put Integer inside. If that is null, we will be seeing NPE no matter we are using Integer or int in this logic.
This is a private method, and inside this class, we know the value won't be null.